render/egl.c

```diff
@@ -250,26 +263,34 @@ struct wlr_egl *wlr_egl_create(EGLenum platform, void *remote_display,
 			"eglQueryWaylandBufferWL");
 	}
 
+	const char *client_exts_str = eglQueryString(EGL_NO_DISPLAY, EGL_EXTENSIONS);
+	if (client_exts_str == NULL) {
+		if (eglGetError() == EGL_BAD_DISPLAY) {
+			wlr_log(WLR_ERROR, "EGL_EXT_client_extensions not supported");
+		} else {
+			wlr_log(WLR_ERROR, "Failed to query EGL client extensions");
+		}
+		return false;
+	}
+
 	const char *device_exts_str = NULL;
 	if (check_egl_ext(client_exts_str, "EGL_EXT_device_query")) {
 		load_egl_proc(&egl->procs.eglQueryDisplayAttribEXT,
```

Comment:
Maybe we can move this `load_egl_proc` call into `egl_create` as well, so that we load procs for client extensions all in the same spot?

Reply:
Sure!